### ISSUE TITLE
False positive for react/display-name

### ISSUE THREAD

**Issue by veloek** [jsx-eslint/eslint-plugin-react#3289]
There seems to be a regression in version 7.30.0. Functions like these are now treated like components and trigger the react/display-name rule:
```
export const demo = (a?: string) => (b?: string) => {
    if (a == null) return null;
    return b;
}
```
I understand that it might be hard to separate function components from regular functions, but aren't React components always capitalized? Couldn't that be a way to identify them?

This wasn't a problem in version 7.29.4 of eslint-plugin-react.

eslint@8.15.0
eslint-plugin-react@7.30.0

**Comment by dottorblaster** [comment]
Plus one. You can see the example here. https://github.com/trento-project/web/runs/6502507233

**Comment by ljharb** [comment]
cc @golopot; perhaps some of your refactors broke this?

**Comment by golopot** [comment]
Yes PR #3276 broke this.

**Comment by GuillaumeAmat** [comment]
Friendly question: Do you plan to make a new release in the coming days?

cc @golopot @ljharb

**Comment by ljharb** [comment]
@GuillaumeAmat i don't have any specific plans for any release. There will be a release when it's ready.
